### DISCUSSION TITLE
Support for screenshots (and video) when running playwright engine in AWS Fargate

### DISCUSSION THREAD

**Discussion by joshuscurtis** [artilleryio/artillery#2524] [Artillery]
Could we explore the feasibility of incorporating screenshot and video support within AWS Fargate environments? This feature could significantly enhance debugging capabilities, especially when discrepancies arise between local testing environments and Fargate deployments.

One potential approach could involve implementing a debug mode where a single VU's journey is recorded. In this mode, screenshots (and videos) of the VU's interactions could be automatically captured and stored in an S3 bucket for subsequent retrieval and analysis. 

This functionality would allow for a close examination of the behavior of applications running in Fargate, aiding in the resolution of discrepancies and ensuring consistent performance across local and Fargate execution.

**Comment by hassy** [comment]
We're planning to add this @joshuscurtis but no ETA yet.

**Comment by joshuscurtis** [comment]
Hey @hassy,
Any updates on when we might be expecting this functionality - or maybe recording HAR/screenshots/video files on an error being thrown?

**Reply by awagholikar-gfm** [reply]
Any updates on this adding this feature ? It has been around 2 years since last comment on thread ?
I +1 this feature.

**Comment by Lewiscowles1986** [comment]
I would like to know more about this also.
While I do think load testing is an inappropriate place to generate videos and screenshots (they should mostly be the same); and reviewing video at scale may be harder than reviewing structured logs of introspected state; I'm interested in perhaps using artillery to coordinate things that might download files, which can be automated at scale, so the playwright feature of recording a video, or snapping an image uploading, directionally aligns there.
